caleb-lab/super-encode-me.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const Tree = require('./models/tree.js')
+const Promise = require('bluebird')
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'})
+let htmlTree = new Tree()
+// module.exports = function(router){
+//
+// }
+
+// let normalDocument = () => {
+fs.readFile(`${__dirname}/assets/minimal.html`, 'utf-8', function(err, data){
+  if(err) Promise.reject(new Error('Can\'t read anything'))
+  let result = []
+  data.split('\n').forEach(node => result.push(node))
```

Comment:
Much work to do. For example, you probably want to handle the following 'ticket items':
1. Validate that it's an HTML doc (you could do this with the doctype tag) (also, this is optional)
2. Parse an opening tag in the buffer string
3. Parse the tag names
4. Add that tag to the tree 
5. Any time you encounter a closing tag, you're stepping back up to the parent node.